pyiso/ieso.py

```diff
@@ -0,0 +1,649 @@
+                                  predisp_const_totals_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], predisp_const_totals_handler.latest_available_datetime())
+                self._get_report_range(result_ts=load_ts, report_handler=predisp_const_totals_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.load, range_start=range_start,
+                                       range_end=range_end)
+        else:
+            LOGGER.warn('No valid options were supplied.')
+        return load_ts
+
+    def get_trade(self, latest=False, yesterday=False, start_at=None, end_at=None, **kwargs):
+        trade_ts = list([])
+        self.handle_options(latest=latest, yesterday=yesterday, start_at=start_at, end_at=end_at, **kwargs)
+        inter_sched_flow_handler = IntertieScheduleFlowReportHandler(ieso_client=self)
+        adequacy_handler = AdequacyReportHandler(ieso_client=self)
+
+        if self.options.get('latest', False):
+            self._get_latest_report_trimmed(result_ts=trade_ts, report_handler=inter_sched_flow_handler,
+                                            parser_format=IESOClient.PARSER_FORMATS.trade)
+        elif self.options.get('start_at', None) and self.options.get('end_at', None):
+            if self.options.get('historical', False) or self.options.get('current_day', False):
+                range_start = max(self.options['start_at'], inter_sched_flow_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], inter_sched_flow_handler.latest_available_datetime())
+                self._get_report_range(result_ts=trade_ts, report_handler=inter_sched_flow_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.trade, range_start=range_start,
+                                       range_end=range_end)
+            if self.options.get('forecast', False):
+                range_start = max(self.options['start_at'], inter_sched_flow_handler.latest_available_datetime(),
+                                  adequacy_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], adequacy_handler.latest_available_datetime())
+                self._get_report_range(result_ts=trade_ts, report_handler=adequacy_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.trade, range_start=range_start,
+                                       range_end=range_end)
+        else:
+            LOGGER.warn('No valid options were supplied.')
+        return trade_ts
+
+    def get_lmp(self, latest=False, yesterday=False, start_at=None, end_at=None, **kwargs):
+        raise NotImplementedError('The IESO does not use locational marginal pricing. See '
+                                  'https://www.oeb.ca/oeb/_Documents/MSP/MSP_CMSC_Report_201612.pdf for details.')
```

Comment:
The closest thing Ontario has to locational marginal pricing (LMP) is the Hourly Ontario Energy Price (HOEP) (e.g. http://reports.ieso.ca/public/DispUnconsHOEP/PUB_DispUnconsHOEP_20170706.xml ) which is an hourly arithmetic average of the 12 five-minute interval energy prices in the Ontario Zone.

Would this be relevant data for the purpose of `get_lmp(...)` or is it more straightforward to leave it unimplemented?